File: features/rolemanagement/src/main/java/com/nivasafinance/features/rolemanagement/permission/service/PermissionReadService.java
package com.nivasafinance.features.rolemanagement.permission.service;

import com.nivasafinance.features.rolemanagement.permission.dto.PermissionResponse;

import java.util.List;

public interface PermissionReadService {
    List<PermissionResponse> getPermissionsByRoles(List<String> roleName);
}

File: features/rolemanagement/src/main/java/com/nivasafinance/features/rolemanagement/permission/service/PermissionReadService.kt
package com.nivasafinance.features.rolemanagement.permission.service

import com.nivasafinance.features.rolemanagement.permission.dto.PermissionResponse

interface PermissionReadService {
    fun getPermissionsByRoles(roleName: List<String>): List<PermissionResponse>
    // interally call role permission mapping
    // internally call role permission group service
    // internall call PermissionReptository to get permissions by permisson ids (list)
}
